server/controllers/games.ts

```diff
@@ -14,52 +39,61 @@ async function createGame(req, res) {
       price_per_head: data.price_per_head,
       contact_details: data.contact_details
     });
+
     res.status(201).json(game)
   } catch (error) {
     console.error('Error creating game:', error);
     res.status(500).json({ message: `Internal server issue: ${error}` })
   }
 }
 
-async function getGames(req, res) {
+async function getGames(req: Request, res: Response): Promise<void> {
   try {
-    const games = await Game.find().populate('venue').populate('players')
-    res.status(200).json(games)
+    const games = await Game.find().populate('venue').populate('players');
+    res.status(200).json(games);
   } catch (error) {
     console.error('Error fetching games:', error);
-    res.status(500).json({ message: `Internal server issue: ${error}` })
+    res.status(500).json({ message: `Internal server issue: ${error}` });
   }
 }
 
-async function getGame(req, res) {
+
+
+async function getGame(req: Request, res: Response): Promise<void> {
   try {
-    const gameId = req.params.gameId
-    const game = await Game.findById(gameId).populate('venue').populate('players')
-    res.status(200).json(game)
+    const { gameId } = req.params;
+    const game = await Game.findById(gameId).populate('venue').populate('players');
+    if (!game) {
+      res.status(404).json({ message: "Game not found" });
+      return;
```

Comment:
for now it is the only place with explicit `return;`
lis it on purpose?

Reply:
I would say the purpose is to quit the function when some validation fails (e.g. no game with the id is found)
In controllers/user.ts he does it like this:
`return res.status(400).json({ error: "Invalid username or email" });`
But the effect is the same, right?